src/cdk/table/array-data-source.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DataSource} from './data-source';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Array-based data source that can be used to connect to a CollectionViewer. Will trigger an
+ * update to the CollectionViewer whenever a new data array is set.
+ */
+export class ArrayDataSource<T> implements DataSource<T> {
```

Comment:
Let's chat - I had in mind that we would create an `ArrayDataSource` and another more material-specific `MdTableDataSource` that would be able to accept an instance of MdSort and MdPaginator. I'm open to a different approach as well